### PR TITLE
More tests in destination accessor

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

In the `destination-accessor.spec.ts` we had already two tests regarding the destination lookup considering two combination of the selection strategy ({alwaysSubscriber, alwaysProvider, subscriberFirst}) and provided user token ({subscriberUserToken,providerUserToken,noUser}). This PR adds not additional tests for all possible  9 combinations.

### CODE REVIEW

File: packages/core/test/scp-cf/destination-accessor.spec.ts [hunk 1]
```diff
@@ -1069,4 +1022,192 @@ describe('destination-accessor', () => {
...
+          2,
+          'https://destination.example.com',
+          providerServiceToken,
+          expect.anything()
+        );
+        expect(actualProvider!.url).toBe(subscriberDestination.URL);
+      });
+    });
+
+    describe('providerUserToken x  {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
```
Comment (mr-flannery) [inline packages/core/test/scp-cf/destination-accessor.spec.ts:1164]:
```suggestion
    describe('providerUserToken x {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
```

File: packages/core/test/scp-cf/destination-accessor.spec.ts [hunk 2]
```diff
@@ -1069,4 +1022,192 @@ describe('destination-accessor', () => {
...
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(
+          providerUserJwt,
+          alwaysSubscriber
+        );
+        assertNothingCalledAndNullFound(mocks, actual);
+      });
+
+      it('providerUserToken x alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
```
Comment (mr-flannery) [inline packages/core/test/scp-cf/destination-accessor.spec.ts:1175]:
```suggestion
      it('providerUserToken x alwaysProvider: should not send a request to retrieve remote subscriber destination and return provider destination.', async () => {
```
I think you copy&pasted this typo multiple times.

File: packages/core/test/scp-cf/destination-accessor.spec.ts [hunk 3]
```diff
@@ -1069,4 +1022,192 @@ describe('destination-accessor', () => {
...
+      });
+
+      it('noUserToken && alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(undefined, alwaysProvider);
+        assertSubscriberNotCalledAndProviderFound(mocks, actual!);
+      });
+
+      it('noUserToken x subscriberFirst: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
```
Comment (marikaner) [inline packages/core/test/scp-cf/destination-accessor.spec.ts:1205]: "send" (also above)

Comment (marikaner) [reply packages/core/test/scp-cf/destination-accessor.spec.ts:1205]: what is the difference between `x` and `&&` above? Maybe the descriptions can be slightly improved?

**Pull request merged**